org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/actions/ToggleBreakpointAdapter.java

```diff
@@ -543,15 +567,40 @@ static IStatus doLineBreakpointToggle(ISelection selection, IWorkbenchPart part,
 					breakpoint.setConditionEnabled(true);
 					breakpoint.setConditionSuspendOnTrue(true);
 				}
-
 				BreakpointToggleUtils.setUnsetTracepoints(false);
 			}
+
+			if (BreakpointToggleUtils.isTriggerpoints() && selection instanceof ITextSelection && part instanceof JavaEditor) {
+
+				breakpoint.setTriggerPoint(true);
+				BreakpointToggleUtils.setTriggerpoints(false);
+
+			}
+
+			if (BreakpointToggleUtils.isHitpoint() && selection instanceof ITextSelection && part instanceof JavaEditor) {
+
+				breakpoint.setHitCount(BreakpointToggleUtils.getHitCount());
+				BreakpointToggleUtils.setHitpoints(false);
+
+			}
 			if (locator == null) {
 				new BreakpointLocationVerifierJob(document, parseCompilationUnit(type.getTypeRoot()), breakpoint, lnumber, tname, type, editor, bestMatch).schedule();
 			}
 			if (BreakpointToggleUtils.isToggleTracepoints()) {
 				BreakpointToggleUtils.setUnsetTracepoints(false);
 			}
+
+			if (BreakpointToggleUtils.isTriggerpoints()) {
+
+				BreakpointToggleUtils.setTriggerpoints(false);
+
+			}
+
+			if (BreakpointToggleUtils.isHitpoint()) {
+
+				BreakpointToggleUtils.setHitpoints(false);
```

Comment:
this "if" block should be merged with the previous "if" block with **more** conditions but similar code inside